### PR TITLE
Release 1.2.0

### PR DESCRIPTION
**Pull request by dylanuys**
- Rewards will now be  based on historical performance by computing the squared ROC AUC of up to the last 100 predictions made by miners
- Scoring will now use a lower alpha (0.01) in the exponential moving average to give additional weight to historical performance

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -19,8 +19,9 @@ The [BitMind platform](https://app.bitmindlabs.ai/) offers a best-in-class devel
 
 ## Quick Links
 
-- [Mining Guide 🔨](docs/Mining.md)
+- [Mining Guide ⛏️](docs/Mining.md)
```
Comment (kenobijon) [inline README.md:22]: ty, missed this previously!

File: bitmind/base/validator.py [hunk 2]
```diff
@@ -375,18 +378,27 @@ def save_state(self):
 
         # Save the state of the validator to file.
         np.savez(
-            self.config.neuron.full_path + "/state.npz",
+            os.path.join(self.config.neuron.full_path, "state.npz"),
```
Comment (kenobijon) [inline bitmind/base/validator.py:381]: could this breeak anything? Why was this working previously if it needs to be changed?

File: bitmind/validator/reward.py [hunk 3]
```diff
@@ -21,46 +21,53 @@
...
-            reward = 1. if np.round(pred) == label else 0.
-            reward *= count_penalty(pred)
+            miner_hotkey = axon.hotkey
+            if uid in performance_tracker.miner_hotkeys and performance_tracker.miner_hotkeys[uid] != miner_hotkey:
+                bt.logging.info(f"Miner hotkey changed for UID {uid}. Resetting performance metrics.")
+                performance_tracker.reset_miner_history(uid, miner_hotkey)
+
+            performance_tracker.update(uid, pred_prob, label, miner_hotkey)
+            metrics = performance_tracker.get_metrics(uid, window=100)
+            reward = metrics['auc'] ** 2
```
Comment (kenobijon) [inline bitmind/validator/reward.py:62]: for visibility, why AUC vs linear combination of F-1, MCC, etc. ?

**Review by kenobijon — APPROVED** [review]
conditionally approve upon minor comments

File: bitmind/validator/reward.py (hunk 3 above)
Comment (dylanuys) [reply bitmind/validator/reward.py:62]: 1. robustness to class imbalance
2. observed stability in simulations
3. statistical interpretability of rewards 
4. squared auc to amplify the difference in performance, compressing lower values (0.99 becomes 0.98, 0.5 becomes 0.25)

File: bitmind/base/validator.py (hunk 2 above)
Comment (dylanuys) [reply bitmind/base/validator.py:381]: Won't break anything, tested this specifically - just for consistency with the new code I added to load/save the performance tracker

**Pull request merged**